src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -1150,14 +1150,26 @@ public static void EmitObject(string objectFilePath, IReadOnlyCollection<Depende
                 }
 
                 if (logger.IsVerbose)
-                    logger.LogMessage($"Finalizing output to '{objectFilePath}'...");
+                    logger.LogMessage($"Emitting debug information");
+
+                // Native side of the object writer is going to do more native memory allocations.
+                // Free up as much memory as possible so that we don't get OOM killed.
+                // This is potentially a waste of time. We're about to end the process and let the
+                // OS "garbage collect" the entire address space.
+                // https://github.com/dotnet/runtime/issues/83180
+                GC.Collect();
+                GC.WaitForPendingFinalizers();
+                GC.Collect();
```

Comment:
```suggestion
```
Do we expect to have a lot of finalizable objects holding onto resources? I would not expect this to help much.

Reply:
No, it's a cargo cult